Core/FieldHandler/EzImageAsset.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Kaliop\eZMigrationBundle\Core\FieldHandler;
+
+use eZ\Publish\Core\FieldType\ImageAsset\Value;
+use Kaliop\eZMigrationBundle\API\FieldValueImporterInterface;
+use Kaliop\eZMigrationBundle\API\FieldDefinitionConverterInterface;
+use Kaliop\eZMigrationBundle\Core\Matcher\ContentMatcher;
+
+class EzImageAsset extends AbstractFieldHandler implements FieldValueImporterInterface
+{
+    protected $contentMatcher;
+
+    public function __construct(ContentMatcher $contentMatcher)
+    {
+        $this->contentMatcher = $contentMatcher;
+    }
+
+    /**
+     * Creates a value object to use as the field value when setting an ez image asset field type.
+     *
+     * @param array|string|int $fieldValue The definition of the field value, structured in the yml file
+     * @param array $context The context for execution of the current migrations. Contains f.e. the path to the migration
+     * @return Value
+     */
+    public function hashToFieldValue($fieldValue, array $context = array())
+    {
+        $altText = '';
+
+        if ($fieldValue === null) {
+            return new Value();
+        }
+        if (isset($fieldValue['alt_text'])) {
+            $altText = $fieldValue['alt_text'];
+        }
+
+        if (is_array($fieldValue) && array_key_exists('destinationContentId', $fieldValue)) {
+            // fromHash format
+            $id = $fieldValue['destinationContentId'];
+        } else {
+            // simplified format
+            $id = $fieldValue;
+        }
+
+        if ($id === null) {
+            return new Value();
+        }
+
+        // 1. resolve relations
+        $id = $this->referenceResolver->resolveReference($id);
+        // 2. resolve remote ids
+        $id = $this->contentMatcher->matchOneByKey($id)->id;
+
+        return new Value($id, $altText);
```

Comment:
Are you sure that the $id is int? I had similar issues when using references https://github.com/ezsystems/ezplatform-ee-demo/blob/v2.5.1/src/AppBundle/Migration/FieldHandler/EzImageAsset.php#L37

is your solution works with content exporter command?

Reply:
ping @florianalexandre any answer?

Reply:
Any plans to merge this?